### PR TITLE
Use Druid's extension loading for integration test instead of maven

### PR DESCRIPTION
**Pull request by jihoonson**
### Description

Currently, our integration test uses maven to pull in dependencies for some extensions instead of Druid's extension loading system. The discrepancy in how Druid loads extension dependencies between in production and in integration tests makes our integration tests less trust-able especially when it comes to class loading. This PR fixes it to use Druid's extension system for every extension. 

This PR also cleans up a bit how integration test configuration system works. 

- Before this PR, some integration tests use a fixed loadList that is set in `setupData` in `druid.sh`. This loadList is not overrideable which is confusing and also hard to debug. This PR removes the fixed loadList.
- `docker-compose.override-env.yml` is deleted. Every docker-compose file should support override-configs now.
- An empty `override.config.path` is allowed now. `environment-configs/empty-config` will be used in that case.
- This PR also removes unnecessary docker compose settings for `custom-coordinator-duties` and `shuffle-deep-store` groups. The docker compose setting should be used only when the test group requires extra nodes besides the default nodes defined in `docker-compose.yml`.

<hr>

##### Key changed/added classes in this PR
 * `copy_resources_template.sh`
 * `druid.sh`

<hr>

This PR has:
- [x] been self-reviewed.

### CODE REVIEW

File: integration-tests/script/copy_resources_template.sh [hunk 1]
```diff
@@ -27,9 +27,16 @@ cp -r client_tls docker/client_tls
 rm -rf $SHARED_DIR/docker
 mkdir -p $SHARED_DIR
 cp -R docker $SHARED_DIR/docker
-mvn -B dependency:copy-dependencies -DoutputDirectory=$SHARED_DIR/docker/lib
 
-# Make directories if they dont exist
+pushd ../
+rm -rf apache-druid-$DRUID_VERSION
+mvn -DskipTests -T1C -Danimal.sniffer.skip=true -Dcheckstyle.skip=true -Ddruid.console.skip=true -Denforcer.skip=true -Dforbiddenapis.skip=true -Dmaven.javadoc.skip=true -Dpmd.skip=true -Dspotbugs.skip=true install -Pdist
+tar xzf distribution/target/apache-druid-$DRUID_VERSION-bin.tar.gz
```
Comment (clintropolis) [inline integration-tests/script/copy_resources_template.sh:34]: it looks like maven assembly plugin supports a 'dir' format, which "Creates an exploded directory format" according to the docs here https://maven.apache.org/plugins/maven-assembly-plugin/assembly.html. What do you think about making a custom integration-tests version of https://github.com/apache/druid/blob/master/distribution/src/assembly/assembly.xml (which leaves out all of the normal distribution stuff like examples and licenses and stuff) along with custom distribution build profile so we could skip making and then unpacking the archive? I suspect it would shave off some overhead, anecdotally it seems like this part is pretty slow when i do these things locally.

semi related, (but not to this PR) the normal docker image could probably stand to do this too https://github.com/apache/druid/blob/master/distribution/docker/Dockerfile#L43, but it would need all the normal distribution stuff so it would just need a way to override the format or just like make the 'dir' version the base and then a 'tar' version that depends on the output of the base version to archive it for a distribution.

**Review by clintropolis — COMMENTED** [review]
overall lgtm, seems much safer for integration tests to more closely reflect an actual installation

File: integration-tests/script/copy_resources_template.sh (hunk 1 above)
Comment (jihoonson) [reply integration-tests/script/copy_resources_template.sh]: That's a good idea, though the extra time that this PR added in each integration test was about 2-5 min in travis which seems not that bad. It sounds better since it would be clearer. I added a new maven profile, `integration-test`, that creates an exploded directory under `distribution/target/`. Integration tests now use this new profile. In a simple test in my machine, the `integration-test` profile took 54 sec while the `dist` profile took 90 sec.

File: distribution/src/assembly/integration-test-assembly.xml [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+        </fileSet>
+
+        <fileSet>
+            <directory>${project.build.directory}/hadoop-dependencies</directory>
+            <includes>
+                <include>*/*/*</include>
+            </includes>
+            <outputDirectory>hadoop-dependencies</outputDirectory>
+        </fileSet>
+        <fileSet>
```
Comment (clintropolis) [inline distribution/src/assembly/integration-test-assembly.xml:45]: are all of the example files needed?

File: integration-tests/docker/docker-compose.yml [hunk 3]
```diff
@@ -36,6 +36,7 @@ services:
     environment:
       - DRUID_INTEGRATION_TEST_GROUP=${DRUID_INTEGRATION_TEST_GROUP}
     depends_on:
+      - druid-overlord
```
Comment (clintropolis) [inline integration-tests/docker/docker-compose.yml:39]: why this change?

**Review by clintropolis — APPROVED** [review]

File: distribution/src/assembly/integration-test-assembly.xml (hunk 2 above)
Comment (jihoonson) [reply distribution/src/assembly/integration-test-assembly.xml:45]: No, I was a bit lazy :sweat_smile: Removed unnecessary files now.

File: integration-tests/docker/docker-compose.yml (hunk 3 above)
Comment (jihoonson) [reply integration-tests/docker/docker-compose.yml:39]: So, I copied this change over from the previous `docker-compose.override-config.yml` file. I'm not sure how it worked before, but now it fails without this change due to circular dependencies in this docker compose file.

**Comment by jihoonson** [comment]
@clintropolis thanks for the review!
